File: python/mirage/commands/builtin/wandb/io.py
from mirage.commands.builtin.generic_bind import CommandIO
from mirage.core.wandb.read import read, read_stream
from mirage.core.wandb.readdir import readdir
from mirage.core.wandb.stat import stat

IO = CommandIO(readdir=readdir,
               read_bytes=read,
               read_stream=read_stream,
               stat=stat,
               is_mounted=lambda a: True,
               local=False,
               max_du_entries=1000)

File: typescript/packages/core/src/commands/builtin/wandb/io.ts
import type { WandbAccessor } from '../../../accessor/wandb.ts'
import { read, readStream } from '../../../core/wandb/read.ts'
import { readdir } from '../../../core/wandb/readdir.ts'
import { stat } from '../../../core/wandb/stat.ts'
import type { CommandIO } from '../generic_bind/index.ts'
export const WANDB_IO: CommandIO<WandbAccessor> = {
  readdir,
  readBytes: read,
  readStream,
  stat,
  isMounted: () => true,
  local: false,
  maxDuEntries: 1000,
}
